src/Makefile.test.include

```diff
@@ -613,6 +613,31 @@ if EMBEDDED_UNIVALUE
 	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -C univalue check
 endif
 
+if ENABLE_BENCH
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY)
+endif
+else
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY)
+endif
+endif
+	@echo "Using the valgrind memory error detector: expect a 10-50x slowdown and ~2x memory usage. valgrind 3.14 or later required."
+	@echo "Running $(TEST_BINARY) under valgrind -- this will take a a while..."
+	valgrind --suppressions=$(top_builddir)/contrib/valgrind.supp --gen-suppressions=all --exit-on-first-error=yes --error-exitcode=1 --quiet $(TEST_BINARY)
+if ENABLE_BENCH
```

Comment:
Retested this PR and wonder if it would not be better to remove the --quiet flags... `make check` isn't quiet either, `make check-valgrind` takes a long time (~90 minutes) to run so some output is reassuring, and the additional output is useful:
```
Using the valgrind memory error detector: expect a 10-50x slowdown and ~2x memory usage. valgrind 3.14 or later required.
Running test/test_bitcoin under valgrind -- this will take a a while...
valgrind --suppressions=../contrib/valgrind.supp --gen-suppressions=all --exit-on-first-error=yes --error-exitcode=1 test/test_bitcoin
==36831== Memcheck, a memory error detector
==36831== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==36831== Using Valgrind-3.16.0.GIT and LibVEX; rerun with -h for copyright info
==36831== Command: test/test_bitcoin
==36831== 
Running 391 test cases...
```

Reply:
Adjusted accordingly :)